### PR TITLE
fix(Datalist): keep user filter when suggestion changes

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

Datalist filter is reset when the suggestions changes
Backslashing the input:
https://user-images.githubusercontent.com/58977230/172173518-4f4c2b10-0a7f-4e28-9a5e-111ca037373c.mov

**What is the chosen solution to this problem?**

Update the filter only if the user did not change it

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Datalist/Datalist.component.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ function Datalist(props) {
 	// Current persisted value
 	// In case of simple values, this is a string
```
Comment (Benrajalu) [inline packages/components/src/Datalist/Datalist.component.js:114]: This does not seem possible anymore

File: packages/components/src/Datalist/Datalist.component.js [hunk 2]
```diff
@@ -145,9 +145,10 @@ function Datalist(props) {
 		}
 
 		if (entry.value !== value) {
-			setValue(entry.value);
+			setEntry(entry);
 		}
-		if (entry.name !== filterValue) {
+		// Update the input value only if user did not change it
+		if (!name || name === filterValue) {
```
Comment (Benrajalu) [inline packages/components/src/Datalist/Datalist.component.js:151]: filterValue filters on `name`, not on `value`...

Comment (gjou-tlnd) [reply packages/components/src/Datalist/Datalist.component.js:150]: yes, value is the technical id, name is the label (which you use in the filter input).. don't blame me, the new DS version of the component will be way better

File: packages/components/src/Datalist/Datalist.stories.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import React from 'react';
+import React, {useState} from 'react';
```
Comment (sgangnant) [inline packages/components/src/Datalist/Datalist.stories.js:1]:
```suggestion
import React, { useState } from 'react';
```

**Review by sgangnant — APPROVED** [review]

**Review by Benrajalu — APPROVED** [review]

**Pull request merged**